src/runtime_src/core/edge/user/device_linux.cpp

```diff
@@ -119,6 +119,17 @@ struct is_ready
   }
 };
 
+struct is_ready_msg
+{
+  using result_type = query::is_ready::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    return std::vector();
+  }
+};
+
```

Comment:
Is this added just for completeness to get some default behavior?

If you don't what to use try/catch for no_such_key with all usages, and if default behavior makes sense, then maybe implement default in the query_request struct itself and remove this code?   

Or even better, you should wrap the query_request in the tools to hide the try/catch for default behavior.  What is default behavior for tools, may not be default behavior for all uses of the query request and if is_ready_msg is not supported, it is not supported :-)

Reply:
Added a wrapper named `device_query_default` that does the trick! It can be found in device.h:516. There are a few calls to that function in info_vmr.cpp which you can look at if you are interested